marlowe-playground-client/src/Marlowe/Blockly.purs

```diff
@@ -962,90 +963,107 @@ caseDefinition g block =
       Nothing -> Either.Left "Could not parse ActionType in caseDefinition function"
       Just actionType -> blockDefinition actionType g block
 
-casesDefinition :: Generator -> Block -> Either String (Array Case)
+casesDefinition :: Generator -> Block -> Either String (Array (Term Case))
 casesDefinition g block = traverse (caseDefinition g) (getAllBlocks block)
 
-boundDefinition :: Generator -> Block -> Either String Bound
+boundDefinition :: Generator -> Block -> Either String (Term Bound)
 boundDefinition g block = do
   from <- parse Parser.bigInteger =<< getFieldValue block "from"
   to <- parse Parser.bigInteger =<< getFieldValue block "to"
-  pure (Bound from to)
+  pure (mkDefaultTerm (Bound from to))
 
-boundsDefinition :: Generator -> Block -> Either String (Array Bound)
+boundsDefinition :: Generator -> Block -> Either String (Array (Term Bound))
 boundsDefinition g block = traverse (boundDefinition g) (getAllBlocks block)
 
-instance hasBlockDefinitionAction :: HasBlockDefinition ActionType Case where
+parseToValue' :: forall a. Parser a -> Parser (Term a)
+parseToValue' p = do
+  v <- p
+  pure $ Term v { row: 0, column: 0 }
+
+statementToTerm :: forall a. Generator -> Block -> String -> Parser a -> Either String (Term a)
+statementToTerm g block name p = case statementToCode g block name of
+  Either.Left _ -> pure $ Hole name Proxy { row: 0, column: 0 }
+  Either.Right s ->
+    (flip parse) s do
+      v <- p
+      pure $ Term v { row: 0, column: 0 }
+
```

Comment:
This is `parse (parseToValue' p) s`, right?

Reply:
indeed